tests/dtm/test_rgw_s3_process_restart.py

```diff
@@ -291,3 +291,139 @@ def test_delete_during_rgw_svc_restart(self):
+
+        que = multiprocessing.Queue()
+
+        self.log.info("Step 1: Start Write Operations for parallel reads "
+                      "and parallel deletes during rgw restart:")
+        write_proc = []
+        for _ in range(0, 2):
+            proc = multiprocessing.Process(target=self.dtm_obj.perform_write_op,
+                                           args=(bucket_name, object_prefix,
+                                                 self.test_cfg['clients'],
+                                                 self.test_cfg['samples'],
+                                                 self.test_cfg['size'],
+                                                 log_file_prefix,
+                                                 que))
+            proc.start()
+            write_proc.append(proc)
+
+        self.log.info("Step 2: Wait for Write Operation to complete.")
+        workload_info_list = []
+        for each in write_proc:
+            if each.is_alive():
+                each.join()
+            resp = que.get()
+            assert_utils.assert_true(resp[0], resp[1])
+            workload_info_list.append(resp[1])
+
+        self.log.info("Step 3: Perform Write,Read,Delete Parallely during rgw restart")
+        parallel_proc = []
+        self.log.info("Step 3a: Start Write in new process")
+        proc_write_op = multiprocessing.Process(target=self.dtm_obj.perform_write_op,
+                                                args=(bucket_name, object_prefix,
+                                                      self.test_cfg['clients'],
+                                                      self.test_cfg['samples'],
+                                                      self.test_cfg['size'],
+                                                      log_file_prefix,
+                                                      que))
+        proc_write_op.start()
+        parallel_proc.append(proc_write_op)
+        self.log.info("Step 3b: Start Read in new process")
+        # Reads, validate and delete
+        proc_read_op = multiprocessing.Process(target=self.dtm_obj.perform_ops,
+                                               args=(workload_info_list[0], que,
+                                                     False,
+                                                     True,
+                                                     True))
+        proc_read_op.start()
+        parallel_proc.append(proc_read_op)
+        self.log.info("Step 3c: Start Deletes in new process")
+        # delete
+        proc_delete_op = multiprocessing.Process(target=self.dtm_obj.perform_ops,
+                                                 args=(workload_info_list[1], que,
+                                                       True,
+                                                       False,
+                                                       False))
+        proc_delete_op.start()
+        parallel_proc.append(proc_delete_op)
```

Comment:
This will do bulk delete.

Reply:
Performs ops performs delete operation by list and deleting object, we are not using the delete from s3bench.